subprojects/gradle/runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/KubernetesApi.kt

```diff
@@ -0,0 +1,153 @@
+package com.avito.android.runner.devices.internal.kubernetes
+
+import com.avito.instrumentation.internal.reservation.adb.waitForCondition
+import com.avito.logger.LoggerFactory
+import com.avito.logger.create
+import io.fabric8.kubernetes.api.model.Pod
+import io.fabric8.kubernetes.api.model.apps.Deployment
+import io.fabric8.kubernetes.client.KubernetesClient
+
+internal interface KubernetesApi {
+    val namespace: String
+    suspend fun deletePod(podName: String): Boolean
+    suspend fun getPodLogs(podName: String): String
+    suspend fun getPod(podName: String): Pod?
+    suspend fun getPodDescription(podName: String): String
+    suspend fun deleteDeployment(deploymentName: String)
+    suspend fun createDeployment(deployment: Deployment)
```

Comment:
This API exposes `io.fabric8.kubernetes` models. 
Do you want to hide them in the future or this abstraction is enough?